store/postgres/src/relational_queries.rs

```diff
@@ -1291,24 +1293,50 @@ impl<'a> QueryFragment<Pg> for InsertQuery<'a> {
         out.push_sql(self.table.qualified_name.as_str());
 
         out.push_sql("(");
-        for column in self.table.columns.iter() {
-            if self.entity.contains_key(&column.field) {
-                out.push_identifier(column.name.as_str())?;
-                out.push_sql(", ");
+
+        // Build the column name list using the subset of all keys among our entities.
+        let unique_column_names: Vec<&Column> = {
+            let mut btreemap = BTreeMap::new();
```

Comment:
This should just be a set; I even think that a `HashSet` is good enough here since we don't care about the order of columns.

Reply:
I ran into some weird ordering errors when using HashSet, which I got rid of when I sorted the resulting vector, so I ended up using a BTreeMap.

The errors were caught by (flaky) results on the `conflicting_implementors_id`. I'm not sure why, though, but I can dig into that.

Reply:
Moving the unique column scanning outside of the `walk_ast` did the trick, but I still have to use `HashMap<&str, &Column>` since the `Column` type does not implement the `Hash` or `Eq` traits.